x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -479,6 +491,48 @@ public String getConcreteIndexName() {
         return state.concreteIndexName;
     }
 
+    public boolean areReservedRolesIndexed() {
+        return areReservedRolesIndexed(state.indexedReservedRolesVersionMap);
+    }
+
+    private static boolean areReservedRolesIndexed(Map<String, String> reservedRolesVersionMap) {
+        if (reservedRolesVersionMap == null) {
+            return false;
+        }
+        Map<String, String> reservedRolesVersions = ReservedRolesStore.versionMap();
+        if (reservedRolesVersionMap.keySet().equals(reservedRolesVersions.keySet()) == false) {
+            return false;
+        }
+        for (String roleName : reservedRolesVersions.keySet()) {
+            if (Integer.parseInt(reservedRolesVersions.get(roleName)) > Integer.parseInt(reservedRolesVersionMap.get(roleName))) {
+                return false;
+            }
+        }
```

Comment:
If indexed roles are "newer" compared to the local version, the reserved roles must have been changed in the code, and a newer node indexed them in this mixed cluster.